evm_loader/program/src/instruction.rs

```diff
@@ -226,9 +228,13 @@ impl<'a> EvmInstruction<'a> {
                 EvmInstruction::CreateAccountWithSeed {base, seed, lamports, space, owner}
             },
             5 => {
+                let (collateral_pool_index, rest) = rest.split_at(4);
                 let (from_addr, rest) = rest.split_at(20);
                 let (sign, unsigned_msg) = rest.split_at(65);
-                EvmInstruction::CallFromRawEthereumTX {from_addr, sign, unsigned_msg}
+                let collateral_pool_index =
+                    u32::from_le_bytes(
+                        collateral_pool_index.try_into().map_err(|_| InvalidInstructionData)?);
```

Comment:
For consistency use `collateral_pool_index.try_into().ok().map(u32::from_le_bytes).ok_or(InvalidInstructionData)?;`